src/system/SystemPacketBuffer.h

```diff
@@ -392,7 +390,31 @@ class DLL_EXPORT PacketBuffer : private pbuf
     const uint8_t * ReserveStart() const;
 
     friend class PacketBufferHandle;
-    friend class ::PacketBufferTest;
+    friend class TestSystemPacketBuffer;
```

Comment:
Can we do something else here? This is terrible... maybe some test class that is a friend and exposes any getters that others need. listing every single test class as a friend is ugly.

Reply:
ok, I've used macro `TEST_F_FROM_FIXTURE`, which allows the implementation of test functionality in the base class. It will limit the number of friend classes.